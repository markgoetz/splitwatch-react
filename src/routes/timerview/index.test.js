
import { h } from 'preact';
import TimerView from './index';
import { shallow } from 'preact-render-spy';

describe('TimerView', () => {
  test('Renders correctly with no splits', () => {
    const context = shallow(<TimerView isPlaying startTime={1000} baseTime={2000} />);
    expect(context).toMatchSnapshot();
  });

  test('Renders correctly with splits', () => {
    const context = shallow(<TimerView isPlaying startTime={1000} baseTime={2000} splits={[500, 700]} />);
    expect(context).toMatchSnapshot();
  });
});