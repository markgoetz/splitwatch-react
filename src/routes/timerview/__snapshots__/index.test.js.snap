// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimerView Renders correctly with no splits 1`] = `
preact-render-spy (1 nodes)
-------
<Wrapper align="left">
  <VerticalLayout align="left">
    <TimerComponent
      isPlaying={true}
      startTime={1000}
      baseTime={2000}
    >
    </TimerComponent>
    <SplitList></SplitList>
  </VerticalLayout>
</Wrapper>

`;

exports[`TimerView Renders correctly with splits 1`] = `
preact-render-spy (1 nodes)
-------
<Wrapper align="left">
  <VerticalLayout align="left">
    <TimerComponent
      isPlaying={true}
      startTime={1000}
      baseTime={2000}
      splits={
        Array [
          500,
          700
        ]
      }
    >
    </TimerComponent>
    <SplitList
      splits={
        Array [
          500,
          700
        ]
      }
    >
    </SplitList>
  </VerticalLayout>
</Wrapper>

`;
